Generation/src/approaches/random/Generator.java

```diff
@@ -279,18 +284,67 @@ else if (rule.lhs().equals("<dim>"))
 			return "";
 		}
 
-		final EBNFClause clause = rule.rhs().get(rng.nextInt(rule.rhs().size()));
+		// We must sort the clauses to ensure that multiple iterations, with the same rng seed, produce the same output
+		List<EBNFClause> clauses = new LinkedList<>(rule.rhs());
+		Collections.sort(clauses, Comparator.comparing(EBNFClause::toString));
+		final EBNFClause clause = clauses.get(rng.nextInt(rule.rhs().size()));
 		
 //		System.out.println("clause: " + clause + " " + (clause.isTerminal() ? "T" : "") +  (clause.isRule() ? "R" : "") +  (clause.isConstructor() ? "C" : ""));
 		
 		if (clause.isTerminal())
 		{
+
+			if (clause.toString().equals("string")) {
+				return "%string%";
+			}
+
+			// Check for primitive arrays
```

Comment:
```suggestion
			// Check for primitive arrays.
			// NOTE: the length of the arrays is biased towards small lengths since rng.nextInt() is re-evaluated at every iteration.
```